### ISSUE TITLE
DisallowImplicitArrayCreation false negative when problematic variable is reassigned after

### ISSUE THREAD

**Issue by tominventisbe** [slevomat/coding-standard#513]
I found a minor corner case with `DisallowImplicitArrayCreation` that is not picked up:

```php
// Works
$nootNoot[] = 5;
```

```php
// Does not work.
$nootNoot[] = 5;
$nootNoot = 5;
```

The implicit array creation here is still happening, I'm just undoing it later :smile: .

Very nice sniff, though! This will definitely catch some unintended problems, especially the one in closures where you forget to bind the variable. :+1:

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/9e97d4be0158a3bcb624f082cff4a0e2d827c991
